### PR TITLE
feat: attach persisted UTM to wallet setup completed

### PR DESCRIPTION
**Pull request by grvgoel81**
## **Description**
Attach persisted utm_* and attribution_id to Wallet Setup Completed only when:
* dataCollectionForMarketing === true
* Persisted attribution exists (selectAttributionRecord)
* Record is within ATTRIBUTION_DEFAULT_TTL_MS (checked at emit time)

Where
* ChoosePassword — spread getWalletSetupCompletedAttributionProperties(store.getState()) on 
* WALLET_SETUP_COMPLETED (covers new wallet + social/password flows)
* ImportFromSecretRecoveryPhrase — same helper on WALLET_SETUP_COMPLETED.

Why

Acquisition context on WSC without re-parsing deeplinks at each site, same consent/TTL rules as the rest of attribution

Jira: https://consensyssoftware.atlassian.net/browse/TO-718

## **Changelog**

CHANGELOG entry: null

## **Related issues**

Fixes:

## **Manual testing steps**

```gherkin
Feature: Wallet Setup Completed includes persisted attribution

  Scenario: New wallet with marketing consent and persisted attribution
    Given the user has opted into marketing data collection
    And persisted attribution exists with utm_* / attribution_id and is not expired
    When the user completes new wallet setup on Choose Password (create or social path)
    Then Wallet Setup Completed is emitted with wallet_setup_type, account_type, and the expected utm_* / attribution_id properties

  Scenario: Import wallet with marketing consent and persisted attribution
    Given the user has opted into marketing data collection
    And persisted attribution exists with utm_* / attribution_id and is not expired
    When the user completes import and sets a password
    Then Wallet Setup Completed is emitted with wallet_setup_type "import", account_type imported, and the expected utm_* / attribution_id properties

  Scenario: Marketing consent off
    Given marketing data collection is false or unset
    And persisted attribution exists
    When the user completes Wallet Setup Completed (new or import)
    Then Wallet Setup Completed must not include utm_* or attribution_id from persistence

```

### **After**
* new wallet
 
Screenshot 2026-05-13 at 11 36 53 AM
Screenshot 2026-05-18 at 9 41 37 AM

* imported wallet
 
Screenshot 2026-05-13 at 11 40 36 AM

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I've included tests if applicable
- [x] I've documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I've applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

#### Performance checks (if applicable)

- [ ] I've tested on Android
  - Ideally on a mid-range device; emulator is acceptable
- [ ] I've tested with a power user scenario
  - Use these [power-user SRPs](https://consensyssoftware.atlassian.net/wiki/spaces/TL1/pages/edit-v2/401401446401?draftShareId=9d77e1e1-4bdc-4be1-9ebb-ccd916988d93) to import wallets with many accounts and tokens
- [ ] I've instrumented key operations with Sentry traces for production performance metrics
  - See [`trace()`](/app/util/trace.ts) for usage and [`addToken`](/app/components/Views/AddAsset/components/AddCustomToken/AddCustomToken.tsx#L274) for an example

For performance guidelines and tooling, see the [Performance Guide](https://consensyssoftware.atlassian.net/wiki/spaces/TL1/pages/400085549067/Performance+Guide+for+Engineers).

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

---

> [!NOTE]
> **Medium Risk**
> Changes analytics payloads for `WALLET_SETUP_COMPLETED` in key onboarding flows by conditionally adding persisted acquisition fields, which could affect downstream event consumers and E2E expectations if the gating logic is wrong.
> 
> **Overview**
> `WALLET_SETUP_COMPLETED` analytics now conditionally includes persisted acquisition fields (`utm_*`, `attribution_id`) for both new-wallet and SRP import flows, by spreading selector-provided props in `ChoosePassword` and `ImportFromSecretRecoveryPhrase`.
> 
> Adds `selectWalletSetupCompletedAttributionAnalyticsProps` plus a new helper `getWalletSetupCompletedAttributionAnalyticsProps` that gates attribution attachment on explicit marketing consent and attribution TTL, with unit coverage.
> 
> Updates analytics E2E infrastructure: fixture builder support for preloading attribution/consent, loosens existing expectations to `containProperties`, and adds a smoke spec validating attribution is present on `WALLET_SETUP_COMPLETED` across create, import, and seedless (Google/Apple) onboarding flows.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit e0e470688be4316ac4dd3875b4a8a5d4681614e0. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: tests/helpers/analytics/expectations/import-wallet.analytics.ts [hunk 1]
```diff
@@ -44,7 +44,7 @@ export const importWalletWithMetricsOptInExpectations: AnalyticsExpectations = {
     },
     {
       name: onboardingEvents.WALLET_SETUP_COMPLETED,
-      matchProperties: {
+      containProperties: {
```
Comment (christopherferreira9) [inline tests/helpers/analytics/expectations/import-wallet.analytics.ts]: Is this change intended? Can we extend the analytics check to check the new properties?

File: tests/helpers/analytics/expectations/new-wallet.analytics.ts [hunk 2]
```diff
@@ -44,7 +44,7 @@ export const newWalletWithMetricsOptInExpectations: AnalyticsExpectations = {
     },
     {
       name: onboardingEvents.WALLET_SETUP_COMPLETED,
-      matchProperties: {
+      containProperties: {
```
Comment (christopherferreira9) [inline tests/helpers/analytics/expectations/new-wallet.analytics.ts]: Is this change intended? Can we extend the analytics check to check the new properties?

File: tests/helpers/analytics/expectations/import-wallet.analytics.ts (hunk 1 above)
Comment (grvgoel81) [reply tests/helpers/analytics/expectations/import-wallet.analytics.ts]: Wallet Setup Completed event can now include optional acquisition fields (utm_*, attribution_id) when marketing consent and attribution are available(https://consensyssoftware.atlassian.net/browse/TO-718). Since these fields are conditional, using containProperties avoids flaky tests while still validating the required onboarding fields (wallet_setup_type, new_wallet, account_type). Using matchProperties would require maintaining separate expectation variants for each optional payload combination.

File: tests/helpers/analytics/expectations/new-wallet.analytics.ts (hunk 2 above)
Comment (grvgoel81) [reply tests/helpers/analytics/expectations/new-wallet.analytics.ts]: Wallet Setup Completed event can now include optional acquisition fields (utm_*, attribution_id) when marketing consent and attribution are available(https://consensyssoftware.atlassian.net/browse/TO-718). Since these fields are conditional, using containProperties avoids flaky tests while still validating the required onboarding fields (wallet_setup_type, new_wallet, account_type). Using matchProperties would require maintaining separate expectation variants for each optional payload combination.

File: tests/helpers/analytics/expectations/import-wallet.analytics.ts (hunk 1 above)
Comment (christopherferreira9) [reply tests/helpers/analytics/expectations/import-wallet.analytics.ts]: Each of the modified expectations is only used once and as we try to decrease the amount of E2E in favor of Component View tests, have we considered avoiding adding a new E2E but instead update the existing ones to reflect the `utm_*` changes and keep a strict validation until we need to loosen it for other corner cases?

Comment (grvgoel81) [reply tests/helpers/analytics/expectations/import-wallet.analytics.ts]: baseline WSC is strict again, utm/attribution coverage is merged into the existing wallet analytics specs

**Review by christopherferreira9 — DISMISSED** [review]

File: app/components/Views/ChoosePassword/index.tsx [hunk 3]
```diff
@@ -487,6 +489,10 @@ const ChoosePassword = () => {
         wallet_setup_type: 'new',
         new_wallet: true,
         account_type: accountType,
+        // Fresh getState(): OAuth sets marketing in handlePostWalletCreation before this track.
+        ...selectWalletSetupCompletedAttributionAnalyticsProps(
```
Comment (ieow) [inline app/components/Views/ChoosePassword/index.tsx]: why are we not using hook?

File: app/components/Views/ChoosePassword/index.tsx [hunk 4]
```diff
@@ -487,6 +489,10 @@ const ChoosePassword = () => {
...
+          ReduxService.store.getState(),
```
Comment (ieow) [inline app/components/Views/ChoosePassword/index.tsx]: we should not use ReduxService directly in the component if possible

File: app/components/Views/ImportFromSecretRecoveryPhrase/index.js [hunk 5]
```diff
@@ -15,7 +15,7 @@ import {
   Platform,
 } from 'react-native';
 import { SafeAreaView } from 'react-native-safe-area-context';
-import { connect } from 'react-redux';
+import { connect, useStore } from 'react-redux';
```
Comment (ieow) [inline app/components/Views/ImportFromSecretRecoveryPhrase/index.js]: use useSelector

**Comment by ieow** [comment]
use this selector
selectWalletSetupCompletedAttributionAnalyticsProps

**Review by ieow — APPROVED** [review]
lgtm

**Review by christopherferreira9 — APPROVED** [review]
